test/runtests.jl

```diff
@@ -0,0 +1,109 @@
+using PyPlot, Base.Test, PyCall
+
+#Checking if there's a GUI
+@test typeof(pygui()) == Symbol
+pygui(false)
+pygui(true)
```

Comment:
You can just do `ENV["MPLBACKEND"]="agg"` to use the Agg non-interactive non-GUI backend.  Non-interactive mode will be the default anyway for test scripts, since `julia foo.jl` runs in non-interactive mode.